src/pulse3D/excel_writer.py

```diff
@@ -203,6 +203,12 @@ def write_xlsx(
     start_time = float(start_time)
     end_time = float(end_time)
 
+    # make sure the width and prominence are tuples
+    prominence_factors = (
```

Comment:
This won't work if the user passes in a tuple with a single value, it will turn in into ((val,), (val,)) which is a tuple of tuples. Also you should add some tests for   all the cases as well